client/lib/human-date/index.js

```diff
@@ -41,14 +38,26 @@ export default function humanDate(
 	}
 
 	if ( millisAgo < MILLIS_IN_MINUTE * 60 * 24 * 7 ) {
-		const days = now.diff( dateOrMoment, 'days' );
-		return i18n.translate( '%(days)dd ago', {
+		const days = now.diff( date, 'days' );
+		return translate( '%(days)dd ago', {
 			args: {
 				days: days,
 			},
 			comment: 'example for a resulting string: 4d ago',
 		} );
 	}
 
-	return dateOrMoment.format( dateFormat );
+	return date.format( dateFormat );
+}
+
+export function useHumanDate( date, dateFormat, interval = EVERY_TEN_SECONDS ) {
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+	const [ tick, doTick ] = useReducer( ( t ) => t + 1, 0 );
+
+	useInterval( doTick, interval );
+
+	return useMemo( () => {
+		return getHumanDateString( date, dateFormat, moment, translate );
+	}, [ tick, date, dateFormat, moment, translate ] );
```

Comment:
Note: I saw an eslint warning about `tick` being an unnecessary dependency. In this case it's needed though to trigger a refresh using `useInterval`.

Reply:
Yes, that's expected, `tick` is kind of a fake dependency.

I'm realizing now that it could be converted to a real dependency if the state contained the `now` date instead:
```js
const [ now, setNow ] = useState( () => moment() );
useInterval( () => setNow( moment() ), interval );
return useMemo( () => getHumanDateString( date, now, ... ), [ date, now ] );
```
Here the `getHumanDateString` function no longer calls `now = moment()` internally, as a side effect, but becomes a pure function of its arguments. That works much better with the `useMemo` memoization and the ESLint rules that assume that things are functional.

Reply:
Added here https://github.com/Automattic/wp-calypso/pull/57826/commits/15c8aca0beef16245f64135ca2dc7eabca1ceb71, however, I think the API of `getHumanDateString` is a little odd. It accepts `now` (as `moment` date), `date` (string or `moment`) and then we pass in `moment` as well. Currently thinking of how to best clean that up.

Reply:
You could improve that a little bit by setting `now` not as a `moment()`, but as a `new Date()`. Then the `useHumanDate` hook doesn't use `moment` itself -- it just passes it down.